### PR TITLE
feat: CP support ColorPicker className and style

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support ColorPicker className and style |
| 🇨🇳 Chinese |          - |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 67d1aa1</samp>

This pull request adds a new feature to the `ConfigProvider` component that allows customizing the `ColorPicker` component globally. It introduces a new `colorPicker` prop to the `ConfigProvider` component and passes it to the `ColorPicker` component through the `ConfigContext`. It also updates the `ColorPicker` component to support custom class names and styles, and adds a test case and documentation for the new feature.

### <samp>🤖 Generated by Copilot at 67d1aa1</samp>

*  Add `colorPicker` prop to `ConfigProvider` component to allow setting common props for `ColorPicker` component ([link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR57-R60), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R173), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R295), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R379), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R112), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R114))
*  Add `colorPicker` property to `ConfigConsumerProps` interface to allow `ColorPicker` component to access common props from `ConfigContext` ([link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR133))
*  Destructure `colorPicker` from `ConfigConsumerProps` and apply custom class name and styles to `Popover` and `ColorPickerPanel` components inside `ColorPicker` component ([link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL93-R94), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR125), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL185-R198), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL209-R219))
*  Add test case to verify `colorPicker` prop of `ConfigProvider` component using `ColorPicker` component and `waitFakeTimer` function ([link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149L3-R3), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R11), [link](https://github.com/ant-design/ant-design/pull/43339/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R770-R798))

### CODE REVIEW

File: components/config-provider/__tests__/style.test.tsx [hunk 1]
```diff
@@ -766,4 +767,31 @@ describe('ConfigProvider support style and className props', () => {
...
+        }}
+      >
+        <ColorPicker />
+      </ConfigProvider>,
+    );
+    const element = container.querySelector<HTMLDivElement>('.ant-color-picker-trigger');
+    expect(element).toHaveClass('cp-colorPicker');
+    expect(element).toHaveStyle({ backgroundColor: 'red' });
+    fireEvent.click(element!);
+    await waitFakeTimer();
```
Comment (MadCcc) [inline components/config-provider/__tests__/style.test.tsx:792]: 受控 open 是不是就好了

File: components/color-picker/ColorPicker.tsx [hunk 2]
```diff
@@ -182,10 +184,18 @@ const ColorPicker: CompoundedComponent = (props) => {
...
 
+  const mergedStyle: React.CSSProperties = { ...colorPicker?.style, ...style };
+
   return wrapSSR(
     <Popover
-      style={styles?.popup}
-      overlayInnerStyle={styles?.popupOverlayInner}
+      style={{
+        ...colorPicker?.styles?.popup,
+        ...styles?.popup,
```
Comment (MadCcc) [inline components/color-picker/ColorPicker.tsx:193]: 先不支持 styles，目前还是非标属性

File: components/config-provider/__tests__/style.test.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/config-provider/__tests__/style.test.tsx]: 不需要支持 styles 的话，这块全部删掉了

File: components/color-picker/ColorPicker.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/color-picker/ColorPicker.tsx]: done~

**Review by MadCcc — APPROVED** [review]

**Pull request merged**